scripts/run_social_messaging.py

```diff
@@ -0,0 +1,167 @@
+        sys.exit(1)
+    except KeyboardInterrupt:
+        sys.exit(0)
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+        description='Run social messaging to forward platform messages to social channels.')
+    parser.add_argument(
+        '--from_config',
+        action='store_true',
+        help='Read social channels config from the pocs.yaml and pocs_local.yaml config files.')
+    args = parser.parse_args()
+
+    def arg_error(msg):
+        print(msg, file=sys.stderr)
+        parser.print_help()
+        sys.exit(1)
+
+    # Initialise social sinks to None
+    social_twitter = None
+    social_slack = None
+    
+    if args.from_config:
+        config = load_config(config_files=['pocs'])
+        
+        if 'social_accounts' in config:
+            social_config = config['social_accounts']
+
+            # See which social sink we can create based on config
+
+            # Twitter sink
+            if social_config is not None and 'twitter' in social_config:
+                twitter_config = social_config['twitter']
+                if twitter_config is not None and \
+                        'consumer_key' in twitter_config and \
+                        'consumer_secret' in twitter_config and \
+                        'access_token' in twitter_config and \
+                        'access_token_secret' in twitter_config:
+                    # Output timestamp should always be True in Twitter
+                    # otherwise Twitter will reject duplicate statuses
+                    if 'output_timestamp' in twitter_config:
+                        output_timestamp = twitter_config['output_timestamp']
+                    else:
+                        output_timestamp = True
+
+                    social_twitter = SocialTwitter(twitter_config['consumer_key'],
+                                                    twitter_config['consumer_secret'],
+                                                    twitter_config['access_token'],
+                                                    twitter_config['access_token_secret'],
+                                                    output_timestamp)
+
+            # Slack sink
+            if social_config is not None and 'slack' in social_config:
+                slack_config = social_config['slack']
+                if slack_config is not None and \
+                        'webhook_url' in slack_config:
+
+                    if 'output_timestamp' in slack_config:
+                        output_timestamp = slack_config['output_timestamp']
+                    else:
+                        output_timestamp = False
+
+                    social_slack = SocialSlack(slack_config['webhook_url'],
+                                                    output_timestamp)
+        else:
+            arg_error('social_accounts setting not defined in config.')
+
+    if not social_twitter and not social_slack:
```

Comment:
Let's treat this as normal. That way our startup scripts can always include executing this script, and it will just exit if there are no social accounts specified, with a non-error message such as "No social messaging sinks defined, exiting."

Reply:
Ok sure